django/db/models/sql/compiler.py

```diff
@@ -1863,6 +1904,19 @@ def execute_sql(self, returning_fields=None):
                     )
                 ]
                 cols = [field.get_col(opts.db_table) for field in self.returning_fields]
+            elif isinstance(opts.pk, CompositePrimaryKey):
+                assert len(returning_fields) == 1
```

Comment:
I believe the same applies here: https://github.com/django/django/pull/18056#discussion_r1839229820 . 

This branch is a very special case. It only executes on older SQLite versions, and only if the model has both an `AutoField` and a `CompositePrimaryKey`. This is an intermediate state during migrations, it's impossible to get into this state otherwise.